### DISCUSSION TITLE
Disable "auto animate in viewport"?

### DISCUSSION THREAD

**Discussion by hanneshoepfnerio** [MAGGIx1404/Textify.js#102] [Q&A]
Hello,
thank you very much for your work on this lib.
I'm using it in a context with scrollTrigger and was wondering if there is any chance of stopping the animation to happen when it comes in to view?
I want to use different trigger to textify.show() elements. But so far I did not really understand how to control the moment of animation and how to stop it?

Best

Hannes

**Comment by MAGGIx1404** [comment]
hello, thank for using our lib. 

currently, you can't stop animation in view. but you can control when animate text which are come in view. you can set threshold for set position in view, where text come and animate. we are currently working on this lib. to add new features and custom animation state etc. and again thanks for support.

**Comment by hanneshoepfnerio** [comment]
Hi, than you so much for your answer and also for your work on the lib.
This answers my question if this would be possible or if I'm just not able to figure it out.
To me the docs raised some wrong expectations since there are sections where you .hide() is after initializing. I assumed that this would be a "sleeping" state, but now I understand that this is just a method to trigger events regardless its default functionality.
Looking forward where this project will go to.
Thank you!

**Reply by MAGGIx1404** [reply]
We are working on implement typescript support and make it more light and easy and fully controlable animations engine. Thanks for supporting us

**Comment by eliasu** [comment]
Hi!

I just wanted to support this request from above! It would be very nice, to disable those objectionObservers on initialization, so that you control the timing when you want to `show()` or `hide()` something. 

Maybe something easy like this (on initialization):

```JS
const textObj = new Textify({
        selector: "[data-type='text']",
        observer: false,
});
```

Thanks for you great work anyway ;)

**Reply by MAGGIx1404** [reply]
yes, we will add this feature in library in few days. thanks for your great idea @eliasu

**Reply by eliasu** [reply]
Hey @MAGGIx1404 
Any update on that subject?
